### ISSUE TITLE
GoogleCloudEnterpriseSearchRetriever consistently returns no results

### ISSUE THREAD

**Issue by mattsday** [langchain-ai/langchain#8219]
### System Info

Versions:

```text
langchain==0.0.240
google-cloud-discoveryengine==0.9.1
google-cloud-aiplatform==1.28.1
```

`GoogleCloudEnterpriseSearchRetriever` it consistently returns zero results without error.

Workarounds / Validations attempted:

* If I put invalid values (e.g. an invalid engine) it causes an error
* Searching with these terms from the console works as expected
* Using `discoveryengine_v1beta.SearchServiceClient()` directly works as expected and provides results

### Information

- [ ] The official example notebooks/scripts
- [X] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [X] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

Given the following code:

```python
retriever = GoogleCloudEnterpriseSearchRetriever(
    project_id=my-project-id,
    search_engine_id=my-engine,
    max_documents=3,
)
query = "Where I can fly to Spain?"
result = retriever.get_relevant_documents(query)
print(result)
```

This will always print `[]` no matter how I tweak the search query etc.

### Expected behavior

Would expect at least one search result and the `result` dictionary not to be empty.

**Comment by pepecura** [comment]
Adding on Matt's question - no response is returned even when the filter is empty: Can it be the case that GoogleCloudEnterpriseSearchRetriever in langchain.retrievers does not return the document snippets for the engines built on a "website" yet, and only supports engines built on "unstructured data"?

**Comment by kamikasky** [comment]
@mattsday @pepecura — I also ran into this issue. It's definitely returning document results. If you're looking to return web search results, you can extend the original class and override `_convert_search_response` like such:

```
from langchain.schema import Document
from google.protobuf.json_format import MessageToDict

class GoogleSearchWebRetriever(GoogleCloudEnterpriseSearchRetriever):
    def _convert_search_response(
        self, results
    ):
        documents = []
        for result in results:
            document_dict = MessageToDict(result.document._pb)
            derived_struct_data = document_dict.get("derivedStructData", None)
            if derived_struct_data:
                doc_metadata = document_dict.get("structData", {})
                doc_metadata.update(derived_struct_data)

                chunks = derived_struct_data.get('snippets', [])
                for chunk in chunks:
                    doc_metadata[
                        "source"
                    ] = f"{derived_struct_data.get('link', '')}"
                    doc_metadata["id"] = document_dict["id"]
                    document = Document(
                        page_content=chunk.get("snippet", ""), metadata=doc_metadata
                    )
                    documents.append(document)

        return documents
```

**Comment by mattsday** [comment]
Thank you @kamikasky - this worked well

**Comment by tomaspiaggio** [comment]
I'm having the same problem with unstructured data. I just copied the example on the documentation and it didn't work.
<img width="443" alt="image" src="https://github.com/langchain-ai/langchain/assets/18428646/afec6a12-1545-41ad-a20a-be7e64593db8">

**Comment by tomaspiaggio** [comment]
I just found the problem. In [this function](https://github.com/langchain-ai/langchain/blob/master/libs/langchain/langchain/retrievers/google_cloud_enterprise_search.py#L128) `_convert_structured_search_response` on line 153, the `getattr` function is not working. Here's the fix: 


```python
def _convert_unstructured_search_response(
    self, results: Sequence[SearchResponse]
) -> List[Document]:
    """Converts a sequence of search results to a list of LangChain documents."""
    from google.protobuf.json_format import MessageToDict

    documents: List[Document] = []

    for result in results:
        document_dict = MessageToDict(
            result.document._pb, preserving_proto_field_name=True
        )

        derived_struct_data = document_dict.get("derived_struct_data", None)
        if not derived_struct_data:
            continue

        doc_metadata = document_dict.get("struct_data", {})
        doc_metadata["id"] = document_dict["id"]

        chunk_type = (
            "extractive_answers"
            if self.get_extractive_answers
            else "extractive_segments"
        )

        if chunk_type not in derived_struct_data:
            continue

        for chunk in derived_struct_data[chunk_type]:
            doc_metadata["source"] = derived_struct_data.get("link", "")

            if chunk_type == "extractive_answers":
                doc_metadata["source"] += f":{chunk.get('pageNumber', '')}"

            documents.append(
                Document(
                    page_content=chunk.get("content", ""), metadata=doc_metadata
                )
            )

    return documents
```

I can do a PR with the fix.

**Comment by holtskinner** [comment]
Should this issue be marked as fixed after #10546
